src/main/java/orm/TableEntity.java

```diff
@@ -115,11 +164,24 @@ private List<TableField> extractAllPersistenceFields(Class<ENTITY> entityClass)
             if (!transientAnnotated) {
                 list.add(
                         idAnnotated
-                                ? new TablePrimaryField(declaredField, jpaSettings)
-                                : new TableField(declaredField, jpaSettings)
+                                ? new TablePrimaryField(declaredField, entity, jpaSettings)
+                                : new TableField(declaredField, entity, jpaSettings)
                 );
             }
         }
         return list;
     }
+
+    private E createNewInstanceByDefaultConstructor(Class<E> entityClass) {
```

Comment:
```suggestion
    private E createNewInstance(Class<E> entityClass) {
```

제가 봤을땐 내부 구현은 언제든지 바뀔수도 있고, 내부 구현이 어떻게 된건지 굳이 메소드 명으로 알려야할 이유를 잘모르겠는데 
`ByDefaultConstructor` 노출시키는것이 이유가 궁금해요! 🙏

Reply:
이부분은 쿼리에서 클래스 리터럴을 넘기는 경우 우선적으로 값이 없는 TableEntity를 만드는 함수인데요!
값이 없는 TableEntity를 만들기 위해서 반드시  기본생성자로 객체를 생성하도록 강조하고싶어서 메서드명을 이렇게 만들었습니다.

추후 설계에서도 이부분은 변하지 않을것으로 예상되어서 클래스명을 
createNewInstanceByDefaultConstructor 로 명확히 나타내고 싶었습니다. 
요부분은 ```createNewInstance```로 수정하도록 하겠습니다. :) 

구현하다보니 하면서 JPA가 기본생성자를 반드시 필요로 하는 이유를 알것같네요 ㅎㅎ.

DDL뿐 아니라 select, insert, delete 같은 구문을 처리할 때도 **'값이 없는 TableEntity'** 가 필요했었습니다

```java
ddlQueryBuilder.createTable(DummyEntity.class)
                        .ifNotExist()
                        .build()

List<Person> people = dqlQueryBuilder.selectFrom(Person.class)
        .findAll()
        .fetch(new PersonRowMapper());
```

Reply:
>구현하다보니 하면서 JPA가 기본생성자를 반드시 필요로 하는 이유를 알것같네요 ㅎㅎ.
DDL뿐 아니라 select, insert, delete 같은 구문을 처리할 때도 '값이 없는 TableEntity' 가 필요했었습니다

아하 그렇네요!  그렇다면 `createNewInstanceByDefaultConstructor`도 괜찮을것 같네요 ㅎ